src/LibChorusTests/VcsDrivers/Mercurial/HgVersionBranchingTests.cs

```diff
@@ -127,5 +127,45 @@ public void DoesNewBranchExist_No()
 				Assert.IsFalse(result, "The only branch should be default.");
 			}
 		}
+
+		[Test]
+		public void CanCreateVersionNumberBranch_BackwardCompatibilityTest()
+		{
+			// Setup
+			using(var repoWithFiles = RepositoryWithFilesSetup.CreateWithLiftFile(stestUser))
+			{
+				var integerBranchName = "70000068";
+				repoWithFiles.Synchronizer.SynchronizerAdjunct = new ProgrammableSynchronizerAdjunct(integerBranchName);
+				var branchingHelper = repoWithFiles.Repository.BranchingHelper;
+				Assert.AreEqual(1, branchingHelper.GetBranches().Count(),
+								"Setup problem in test, should be starting with one branch.");
+				// Make a new branch (should technically be on the remote with a different user...)
+				repoWithFiles.ReplaceSomething("nottheoriginal");
+				repoWithFiles.SyncWithOptions(new SyncOptions
+				{
+					DoPullFromOthers = false,
+					CheckinDescription = "version number branch",
+					DoSendToOthers = false
+				});
+
+				const string myVersion = ""; // Hg default branch name
+
+				// SUT
+				string revNum;
+				bool result = branchingHelper.IsLatestBranchDifferent(myVersion, out revNum);
+
+				// Verification
+				Assert.IsTrue(result, "The only branch should be default.");
+				var revision = repoWithFiles.Repository.GetRevision(revNum);
+				Assert.AreEqual(integerBranchName, revision.Branch, "Wrong branch name in new branch.");
```

Comment:
Hmmm, I guess not, since we're verifying that we did NOT get a new branch with the integer name. I'm thoroughly confused after several minutes studying this method, so I think something needs improving...naming, comments,...partly because not seeing this in an IDE I have to try to remember/guess what on earth a SnychronizerAdjunct and its constructor argument are.